elftools/dwarf/structs.py

```diff
@@ -76,6 +76,18 @@ class DWARFStructs(object):
 
         See also the documentation of public methods.
     """
+
+    __StructsCache = {}
+
+    def __new__(cls, little_endian, dwarf_format, address_size, dwarf_version=2):
+        key = (little_endian, dwarf_format, address_size, dwarf_version)
+        if key in cls.__StructsCache:
+            return cls.__StructsCache[key]
+        newinst = super(DWARFStructs, cls).__new__(cls)
```

Comment:
Nit: I believe you should be able to do just `super()...` here

Reply:
I couldn't because Python 2 support was not dropped at that time :-) 
Done.